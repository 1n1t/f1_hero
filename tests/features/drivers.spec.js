const axios = require('axios');

const startServer = require('../../server');
const driversData = require('../seedsData/drivers');
const { populateDriversCollection } = require('../seeds');

const api = axios.create({
	baseURL: 'http://localhost:3002/api'
});

let server;

beforeAll(async () => {
	server = await startServer();
});

afterAll(done => {
	server.close(done);
});

beforeEach(async done => {
	await populateDriversCollection();
	done();
});

describe('GET /api/drivers', () => {
	it('should fetch all drivers', async () => {
		const drivers = await api.get('/drivers').then(res => res.data.drivers);
		const driversSnap = drivers.map(({ code, name }) => ({ code, name }));

		expect(drivers).toHaveLength(driversData.length);
		expect(driversSnap).toMatchSnapshot();
	});
});
