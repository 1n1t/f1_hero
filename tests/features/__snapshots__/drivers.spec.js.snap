// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /api/drivers should fetch all drivers 1`] = `
Array [
  Object {
    "code": "VET",
    "name": "S. Vettel",
  },
  Object {
    "code": "RIC",
    "name": "D. Ricciardo",
  },
  Object {
    "code": "VAN",
    "name": "S. Vandoorne",
  },
]
`;
